kernel/include/ksched.h

```diff
@@ -67,6 +67,7 @@ void z_ready_thread(struct k_thread *thread);
 void z_thread_single_abort(struct k_thread *thread);
 FUNC_NORETURN void z_self_abort(void);
 void z_requeue_current(struct k_thread *curr);
+struct k_thread *z_swap_next_thread(void);
```

Comment:
@andyross this commit broke the SMP sync example on UP2 - identified with bisect... So, currently it's broken on "master" But one of your patches from #32356 seems to fix this breakage again...